relay-server/src/actors/spooler/mod.rs

```diff
@@ -662,6 +666,26 @@ pub struct BufferService {
 impl BufferService {
     /// Set up the database and return the current number of envelopes.
     async fn setup(path: &PathBuf) -> Result<(), BufferError> {
+        if path.is_absolute() {
+            let mut dir = path.clone();
+            dir.pop();
+
+            if !dir.exists() {
+                relay_log::debug!("creating directory for spooling file: {}", dir.display());
+                DirBuilder::new()
+                    .recursive(true)
+                    .create(dir)
+                    .await
+                    .map_err(|e| BufferError::BufferSetupFailed(e.to_string()))?;
```

Comment:
If we just document this behaviour better , would this be also ok? 

But also if that really simplifies the usage , I'm also ok if the relay will create the folders automatically.

Reply:
We will still have to document this behaviour as well, to make sure it's clear for user that all the folders and sub-folders will be created automatically.

Reply:
Documented in the docs PR in the description.